staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -347,60 +439,64 @@ func (cnc *CloudNodeController) initializeNode(ctx context.Context, node *v1.Nod
 	})
 	if err != nil {
 		utilruntime.HandleError(err)
-		return
+		return err
 	}
+	klog.Infof("Initializing node %s with cloud provider", nodeName)
 
-	curNode, err := cnc.kubeClient.CoreV1().Nodes().Get(context.TODO(), node.Name, metav1.GetOptions{})
+	curNode, err := cnc.nodeInformer.Lister().Get(nodeName)
 	if err != nil {
-		utilruntime.HandleError(fmt.Errorf("failed to get node %s: %v", node.Name, err))
-		return
+		utilruntime.HandleError(fmt.Errorf("failed to get node %s: %v", nodeName, err))
+		return err
```

Comment:
I think we want to return an nil error if the node is not found anymore since that means we no longer need to initialize it.
```
if err != nil {
  if apierrors.IsNotFound(err) {
     return nil
  }

  return err
}
```